File: Python Implementation/Bubble_sort.py
def bubble_sort(a):
    l = len(a)-1
    for i in range(l):
        swap = False
        for j in range(0,l-i):
            if a[j]>a[j+1]:
                swap = True
                a[j],a[j+1] = a[j+1],a[j]
        if swap == False:
            break
    return a

a = [5,4,3,2,1]
print(bubble_sort(a))

File: C++ Implementation/bubble_sort.cpp
#include<bits/stdc++.h>

using namespace std;


void bubble_sort(vector<int>&v){

    for(int i=0; i<v.size()-1; i++)
    {
        bool swp = false;
        for (int j=0; j<v.size()-i-1; j++)
        {
            if(v[j]>v[j+1])
            {
                swap(v[j],v[j+1]);
                swp = true;
            }

        }
        if (swp == false){
            break;
        }

    }

}

int main()
{
    vector<int> v = {1,2,3,4,5,6,7,8,9};
    sort(v.begin(),v.end(),greater<int>());
    for(int i=0; i<v.size(); i++)
    {
        cout << v[i] << endl;
    }
    bubble_sort(v);
    for(int i=0; i<v.size(); i++)
    {
        cout << v[i] << endl;
    }


}
